flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -106,9 +113,14 @@ class DataStreamOverAggregate(
         if (overWindow.lowerBound.isUnbounded &&
           overWindow.upperBound.isCurrentRow) {
           createUnboundedAndCurrentRowProcessingTimeOverWindow(inputDS)
+        } // lowerbound is a BasicType and upperbound is PRECEEDING or CURRENT ROW
+        else if (overWindow.lowerBound.getOffset.getType.isInstanceOf[BasicSqlType]
```

Comment:
What is the check `overWindow.lowerBound.getOffset.getType.isInstanceOf[BasicSqlType]` for and why do you check if `overWindow.upperBound.isPreceding`?

Should the condition rather be like this:
```
else if (overWindow.lowerBound.isPreceding() && !overWindow.lowerBound.isUnbounded() && // bounded preceding
  overWindow.upperBound.isCurrentRow() && // until current row
  overWindow.isRows) // is rows window
```